### PR TITLE
Support for MP3

### PR DESCRIPTION
**Pull request by mauricefisher64**
Remove obsolete unit test
fix potential errors writing streams for PNG and RIFF

## Changes in this pull request
_Give a narrative description of what has been changed._

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [x] All applicable changes have been documented.
- [x] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

File: sdk/src/asset_handlers/mp3_io.rs [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+
+        // wrapper to protect output stream from being gobbled
+        let writer = CAIReadWriteWrapper {
+            reader_writer: output_stream,
+        };
+
+        // write new tag to output stream
+        out_tag
+            .write_to(writer, Version::Id3v24)
+            .map_err(|_e| Error::BadParam("could not write to output stream".to_string()))?;
```
Comment (scouten-adobe) [inline sdk/src/asset_handlers/mp3_io.rs:304]: I don't think it's right to call this a `BadParam`. Is there something like an I/O error that we can pass through?

**Review by scouten-adobe — CHANGES_REQUESTED** [review]

File: sdk/src/asset_handlers/mp3_io.rs [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+            let pos = find_bytes(&tag_bytes, &manifests[0])?;
+
+            return Some((pos as u64, manifests[0].len() as u32));
+        }
+    }
+    None
+}
+
+pub struct Mp3IO {
+    #[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (mauricefisher64) [reply sdk/src/asset_handlers/mp3_io.rs:121]: Same as comment above

File: sdk/src/asset_handlers/mp3_io.rs [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+    mp3_format: String,
+}
+
+impl CAIReader for Mp3IO {
+    fn read_cai(&self, input_stream: &mut dyn CAIRead) -> Result<Vec<u8>> {
+        if let Ok(tag) = Tag::read_from(input_stream) {
+            let mut manifests = Vec::new();
```
[the remark this answers is not in the document]
Comment (mauricefisher64) [reply sdk/src/asset_handlers/mp3_io.rs:128]: We could preallocate to size 2 but by spec we have to explicitly check to make sure there is a single manifest.

File: sdk/src/asset_handlers/mp3_io.rs [hunk 4]
```diff
@@ -0,0 +1,480 @@
...
+        Ok(_) => {
+            // just clone
+            input_stream.rewind()?;
+            output_stream.rewind()?;
+            std::io::copy(input_stream, output_stream)?;
+            Ok(())
+        }
+        Err(_) => {
+            input_stream.rewind()?;
+            mp3io.write_cai(input_stream, output_stream, &[1, 2, 3, 4]) // save arbitrary data
```
[the remark this answers is not in the document]
Comment (mauricefisher64) [reply sdk/src/asset_handlers/mp3_io.rs:171]: This function (add_required_frame) checks for an arbitrary manifest (in a frame).  If there is none it makes one.

File: sdk/src/asset_handlers/mp3_io.rs (hunk 1 above)
Comment (mauricefisher64) [reply sdk/src/asset_handlers/mp3_io.rs:304]: Agreed.

File: sdk/src/asset_handlers/mp3_io.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (scouten-adobe) [reply sdk/src/asset_handlers/mp3_io.rs:125]: See #298 which targets this PR.

File: sdk/src/store.rs [hunk 5]
```diff
@@ -3987,24 +3987,6 @@ pub mod tests {
             .starts_with("ClaimDecoding"))
     }
 
-    #[test]
-    fn test_modify_xmp() {
```
Comment (gpeacock) [inline sdk/src/store.rs:3991]: Should this be removed or just put behind the xmp_write feature?

File: sdk/src/asset_handlers/png_io.rs [hunk 6]
```diff
@@ -477,15 +478,18 @@ impl AssetIO for PngIO {
             .open(asset_path)
             .map_err(Error::IoError)?;
 
-        let mut output_stream = std::fs::OpenOptions::new()
-            .read(true)
-            .write(true)
-            .open(asset_path)
-            .map_err(Error::IoError)?;
+        let mut temp_file = Builder::new()
```
Comment (gpeacock) [inline sdk/src/asset_handlers/png_io.rs:481]: Why are we going to temp files at this level? Isn't it better to let the higher levels pass in a temp file?

Comment (mauricefisher64) [reply sdk/src/asset_handlers/png_io.rs:481]: If you look at this api it only takes a destination.  If you fail you have corrupted the source.

**Review by scouten-adobe — APPROVED** [review]

**Pull request merged**